### PR TITLE
switch to BaseTestNext testing

### PR DESCRIPTION
**Pull request by Evizero**
~~don't merge yet~~. I open this PR now so that you have the opportunity to voice concerns along the way. And that I have a chance to adapt early

The purpose of this PR is to improve the testing a bit and also to upgrade to the latest way testing is performed. This will make this package 0.4 or higher.

My intention is to first target both 0.4 and 0.5 before I possibly introduce any 0.5 only features

### CODE REVIEW

**Comment by Evizero** [comment]
Do you think it would be useful to introduce Travis?

**Comment by Evizero** [comment]
Ok, so this would be my first proposal. A cleanup of sorts. You can go ahead merge this if you agree with the changes.

I will start to work on a more efficient loading logic now. (unless you have change requests to this PR of course!)

**Comment by Evizero** [comment]
Thoughts on this?

I will probably be finished with my refactoring later today. Do you want me to use this PR for everything? I initially thought it might be more convenient for you if I create a separate refactor PR after this is merged.

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+# Documentation: http://docs.travis-ci.com/user/languages/julia/
+language: julia
+os:
```
Comment (johnmyleswhite) [inline .travis.yml]: Should we test against Windows? Do we know if this package failed on Windows in the past?

**Comment by johnmyleswhite** [comment]
This is all good by me in terms of design. I'll merge once you feel it's ready.

**Comment by Evizero** [comment]
I think this is ready to merge. Concerning windows. We need appvenor for that, yes? I'll look into it. Have only used travis in the past.

Once this is merged I rebase my other branch and create a WIP, PR for that. I suspect you will be more critical on those things so it would make sense to split this into a separate PR imho

**Comment by johnmyleswhite** [comment]
You're right: I forgot that AppVeyor and Travis are totally separate. Merging now. Thanks for doing this step first as it was very easy to review.

**Pull request merged**